t/10_susedistribution.t

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/perl
+# Copyright @ SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+use Test::Most;
+use testapi;
+use Test::MockModule;
+use Test::MockObject;
+
+subtest 'handle_password_prompt on serial terminal checks wait_serial' => sub {
+    set_var('VIRTIO_CONSOLE', '1');
+    set_var('BACKEND', 'qemu');
+    my $testapi_mocked = Test::MockModule->new('testapi');
+    $testapi_mocked->mock('is_serial_terminal' => sub { return 1 });
+    $testapi_mocked->mock('type_string' => sub { return 'randompass' });
+    $testapi_mocked->noop('send_key', 'assert_screen');
+    $testapi_mocked->mock('wait_serial' => sub { return 'Password:' });
+    require susedistribution;
+    ok susedistribution::handle_password_prompt();
+};
+
+subtest 'handle_password_prompt assert screen on when not serial_terminal' => sub {
+    set_var('BACKEND', 'qemu');
+    my $testapi_mocked = Test::MockModule->new('testapi');
+    $testapi_mocked->mock('is_serial_terminal' => sub { return 0 });
+    $testapi_mocked->mock('type_string' => sub { return 'randompass' });
+    $testapi_mocked->noop('send_key', 'assert_screen');
+    require susedistribution;
```

Comment:
The problem with requiring `susedistribution` a second time is, that it is basically a no-op.
So changing the functions `is_serial_terminal` etc. after the first `require` will not use the new ones.

But as the test is passing, that means, the test is not sufficient enough.
You are only checking the return value of `handle_password_prompt`, but I think it should check more details.

I will play with it a bit and come back to you.

Reply:
Here is my suggestion: https://gist.github.com/perlpunk/8576af61249bdbba51900821170c47f4

Reply:
interesting approach
I did some minor changes without modifying the main logic
*ready for review*